sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1136,6 +1155,63 @@ Exceptions:
   `property::graph::updatable` was not set when the executable graph was created.
 * Throws with error code `invalid` if any node in `nodes` is not part of the
   graph.
+
+|
+[source, c++]
+----
+void
+update(const command_graph<graph_state::modifiable>& graph);
+----
+
+|Updates the executable graph node inputs & outputs from a topologically
+identical modifiable graph. A topologically identical graph is one with the
+same structure of nodes and edges, and the nodes added in the same order to
+both graphs. Equivalent nodes in topologically identical graphs each have the
+same command, targeting the same device. There is the additional limitation that
+to update an executable graph, every node in the graph must be either a kernel
+command or a host task.
+
+The only characteristic that can differ between two topologically identical
+graphs during an update are the arguments to kernel nodes. For example,
+the graph may capture different values for the USM pointers or accessors used
+in the graph. It is these kernels arguments in `graph` that constitute the
+inputs & outputs to update to.
+
```

Comment:
Earlier on in the parent section we say

> The aspects of a kernel execution node that can be configured during update are:
> 
> * Parameters to the kernel.
> * Execution ND-Range of the kernel.

However, if i'm right @Bensuo whole graph update doesn't take into account a different ND-Range in the new kernel nodes (and do so would be an error?) I think we should say something explicitly about the ND-Range of a kernel node changing to avoid confusion.

Reply:
I think we should be able to support execution range update in whole graph update as well, so this wording should probably change.

Reply:
I changed the wording to also include update on the ND-Range. Do you think this is enough or should I be more descriptive?

Reply:
looks good, thanks